diff --git a/src/option.c b/src/option.c
--- a/src/option.c
+++ b/src/option.c
@@ -286,7 +286,7 @@ static void option_parse_token(char ** argv, int * flag_parsed_next, int * requi
             switch(o->type){
               case('p'):{
                 // call the function in the variable
-                void(*fp)() = o->variable;
+                void (*fp)(char *) = o->variable;
                 fp(arg);
                 break;
               }
diff --git a/src/utilities.h b/src/utilities.h
--- a/src/utilities.h
+++ b/src/utilities.h
@@ -33,10 +33,6 @@ extern enum OutputFormat_t outputFormat;  /* format of the output */
 #define MAX_PATHLEN PATH_MAX
 
 
-#ifdef __linux__
 #define ERROR_LOCATION __func__
-#else
-#define ERROR_LOCATION __LINE__
-#endif
 
 #define FAIL(...) FailMessage(rank, ERROR_LOCATION, __VA_ARGS__)
 void FailMessage(int rank, const char *location, char *format, ...);
